depthai_demo.py

```diff
@@ -112,46 +130,69 @@ def print_sys_info(info):
 
     if len(conf.args.report) > 0:
         pm.create_system_logger()
 
-    nn_pipeline = nn_manager.create_nn_pipeline(pm.p, pm.nodes, shaves=conf.args.shaves, use_depth=conf.useDepth,
-                                                use_sbb=conf.args.spatial_bounding_box and conf.useDepth,
-                                                minDepth=conf.args.min_depth, maxDepth=conf.args.max_depth,
-                                                sbbScaleFactor=conf.args.sbb_scale_factor)
-    pm.create_nn(nn=nn_pipeline, sync=conf.args.sync)
+    if conf.useNN:
+        nn_pipeline = nn_manager.create_nn_pipeline(pm.p, pm.nodes, shaves=conf.args.shaves, use_depth=conf.useDepth,
+                                                    use_sbb=conf.args.spatial_bounding_box and conf.useDepth,
+                                                    minDepth=conf.args.min_depth, maxDepth=conf.args.max_depth,
+                                                    sbbScaleFactor=conf.args.sbb_scale_factor)
+        pm.create_nn(nn=nn_pipeline, sync=conf.args.sync)
 
     # Start pipeline
     device.startPipeline(pm.p)
     pm.create_default_queues(device)
-    nn_in = device.getInputQueue(nn_manager.input_name, maxSize=1, blocking=False) if not conf.useCamera else None
-    nn_out = device.getOutputQueue(nn_manager.output_name, maxSize=1, blocking=False)
+    nn_in = device.getInputQueue(nn_manager.input_name, maxSize=1, blocking=False) if not conf.useCamera and conf.useNN else None
+    nn_out = device.getOutputQueue(nn_manager.output_name, maxSize=1, blocking=False) if conf.useNN else None
 
-    sbb_out = device.getOutputQueue("sbb", maxSize=1, blocking=False) if nn_manager.sbb else None
+    sbb_out = device.getOutputQueue("sbb", maxSize=1, blocking=False) if conf.useNN and nn_manager.sbb else None
     log_out = device.getOutputQueue("system_logger", maxSize=30, blocking=False) if len(conf.args.report) > 0 else None
 
+    median_filters = cycle([item for name, item in vars(dai.MedianFilter).items() if name.startswith('KERNEL_') or name.startswith('MEDIAN_')])
+    for med_filter in median_filters:
+        # move the cycle to the current median filter
+        if med_filter == pm.depthConfig.getMedianFilter():
+            break
+
     if conf.useCamera:
-        pv.create_queues(device)
+        def create_queue_callback(queue_name):
+            if queue_name in [Previews.disparity_color.name, Previews.disparity.name, Previews.depth.name, Previews.depth_raw.name]:
+                Trackbars.create_trackbar('Disparity confidence', queue_name, 0, 255, conf.args.disparity_confidence_threshold,
+                         lambda value: pm.update_depth_config(device, dct=value))
+                if queue_name in [Previews.depth_raw.name, Previews.depth.name]:
+                    Trackbars.create_trackbar('Bilateral sigma', queue_name, 0, 250, conf.args.sigma,
```

Comment:
Not sure, is it possible to dynamically change the maximum value on trackbar?
It would be fun to play with extreme values. 65535 is the max for bilateral sigma, 127/8 for LR-check threshold.
Nice to have if it's easy.

Reply:
@szabi-luxonis I didn't want to expand unnecessarily the cli args, so I made them configurable by env variables

```
DISP_CONF_MIN = int(os.getenv("DISP_CONF_MIN", 0))
DISP_CONF_MAX = int(os.getenv("DISP_CONF_MAX", 255))
SIGMA_MIN = int(os.getenv("SIGMA_MIN", 0))
SIGMA_MAX = int(os.getenv("SIGMA_MAX", 250))
LRCT_MIN = int(os.getenv("LRCT_MIN", 0))
LRCT_MAX = int(os.getenv("LRCT_MAX", 10))
```

Reply:
By dynamic I meant if the user sets the slider to max value, increase the max value on the slider itself (or better shift the min max range) , so it can be set to higher than initial